### PR TITLE
Switched to a properties file for the default configuration

### PR DESCRIPTION
**Pull request by bdemers**
This makes the yaml dependency optional for downstream consumers (if they do NOT want to parse ~/.okta/okta.yaml, i.e. on Android)

NOTE: SnakeYaml is still a compile scope dependency, so it would need to be excluded (this keeps things backwards compatible and easier to use for the majority of use cases)

PR NOTE: do not merge until okta-sdk-java the `yaml-optional` branch of okta/okta-sdk-java is merged

### CODE REVIEW

File: impl/src/main/java/com/okta/authn/sdk/impl/client/DefaultAuthenticationClientBuilder.java [hunk 1]
```diff
@@ -98,8 +100,7 @@ else if (SYSPROPS_TOKEN.equalsIgnoreCase(location)) {
                 Resource resource = resourceFactory.createResource(location);
 
                 PropertiesSource wrappedSource;
-                if (Strings.endsWithIgnoreCase(location, ".yaml") ||
-                        Strings.endsWithIgnoreCase(location, ".yml")) {
```
Comment (bretterer) [inline impl/src/main/java/com/okta/authn/sdk/impl/client/DefaultAuthenticationClientBuilder.java]: Will removing `.yml` cause any backwards compat issues?

File: integration-tests/src/test/groovy/com/okta/authn/sdk/its/AuthenticationClientIT.groovy [hunk 2]
```diff
@@ -249,14 +249,14 @@ class AuthenticationClientIT extends AuthenticationTestSupport {
...
-        def response1 = authClient.authenticate(user.profile.getEmail(), USER_PASSWORD, null, ignoringStateHandler)
+        def response1 = authClient.authenticate(user.getProfile().getEmail(), USER_PASSWORD, null, ignoringStateHandler)
 
         assertThat response1.getStatus(), is(AuthenticationStatus.MFA_ENROLL)
         assertThat response1.getSessionToken(), nullValue()
 
         // enroll in TOTP factor
         def totpFactor = authClient.instantiate(TotpFactorProfile)
-                .setCredentialId(user.profile.getEmail())
+                .setCredentialId(user.getProfile().getEmail())
```
Comment (bretterer) [inline integration-tests/src/test/groovy/com/okta/authn/sdk/its/AuthenticationClientIT.groovy]: Should this all be done in another PR?  I think you are doing 2 different things here?   1, changing the prop file, and then 2. this updating user.profile to be `getProfile`

Comment (bdemers) [reply integration-tests/src/test/groovy/com/okta/authn/sdk/its/AuthenticationClientIT.groovy]: moved to #27, I'll update this PR after that one merges
